profiling/src/profile/internal/labels/label_set.rs

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+use super::super::*;
+
+/// A canonical representation for sets of labels.
+/// You should only use the impl functions to modify this.
+#[derive(Clone, Debug, Default, Eq, PartialEq, Hash)]
+pub struct LabelSet {
+    sorted_labels: Box<[LabelId]>,
+}
```

Comment:
Calling this a `LabelSet` seems somewhat weird to me since It's not actually a `Set` data structure, and we end up with weird things such as `FxIndexSet<LabelSet>` to actually use it as a `Set`.

Naming is always hard... Here's a few suggestions:

* `SampleLabels`
* `LabelGroup`

Reply:
I'm not sure if it matters in practice, but `[0, 1, 2]` and `[2, 1, 0]` wouldn't be the same "LabelSet". The number of items in these sets is expected to be small. I'm not sure if it's worth "normalizing" by either:

- Sorting the vec.
- Using a set structure.

But if we do nothing, then I agree with Ivo that it's not really a set, so we should probably avoid that name.

Reply:
Added a comment to make this more obvious